src/main/java/org/ow2/proactive/connector/iaas/cloud/provider/azure/AzureProvider.java

```diff
@@ -324,23 +342,38 @@
+                                                                         .map(VirtualMachineSizeTypes::fromString);
+        VirtualMachine.DefinitionStages.WithCreate creatableVMWithSize = creatableVirtualMachineWithImage.withSize(optionalHardwareType.orElse(DEFAULT_VM_SIZE))
+                                                                                                         .withOSDiskName(createUniqOSDiskName(instanceTag));
 
         // Add init script(s) using dedicated Microsoft extension
         Optional.ofNullable(instance.getInitScript()).map(InstanceScript::getScripts).ifPresent(scripts -> {
             if (scripts.length > 0) {
                 StringBuilder concatenatedScripts = new StringBuilder();
                 Lists.newArrayList(scripts)
                      .forEach(script -> concatenatedScripts.append(script).append(SCRIPT_SEPARATOR));
-                creatableVMWithSize.defineNewExtension(createUniqueScriptName(instanceTag))
-                                   .withPublisher(SCRIPT_EXTENSION_PUBLISHER)
-                                   .withType(SCRIPT_EXTENSION_TYPE)
-                                   .withVersion(SCRIPT_EXTENSION_VERSION)
-                                   .withMinorVersionAutoUpgrade()
-                                   .withPublicSetting(SCRIPT_EXTENSION_CMD_KEY, concatenatedScripts.toString())
-                                   .attach();
+                if (operatingSystemType.equals(OperatingSystemTypes.LINUX)) {
+                    creatableVMWithSize.defineNewExtension(createUniqueScriptName(instanceTag))
+                                       .withPublisher(SCRIPT_EXTENSION_PUBLISHER_LINUX)
+                                       .withType(SCRIPT_EXTENSION_TYPE_LINUX)
+                                       .withVersion(SCRIPT_EXTENSION_VERSION_LINUX)
+                                       .withMinorVersionAutoUpgrade()
+                                       .withPublicSetting(SCRIPT_EXTENSION_CMD_KEY, concatenatedScripts.toString())
+                                       .attach();
+                } else if (operatingSystemType.equals(OperatingSystemTypes.WINDOWS)) {
+                    creatableVMWithSize.defineNewExtension(createUniqueScriptName(instanceTag))
+                                       .withPublisher(SCRIPT_EXTENSION_PUBLISHER_WINDOWS)
+                                       .withType(SCRIPT_EXTENSION_TYPE_WINDOWS)
+                                       .withVersion(SCRIPT_EXTENSION_VERSION_WINDOWS)
+                                       .withMinorVersionAutoUpgrade()
+                                       .withPublicSetting(SCRIPT_EXTENSION_CMD_KEY, concatenatedScripts.toString())
+                                       .attach();
+                } else {
+                    throw new RuntimeException("ERROR Operating System of type '" + operatingSystemType.toString() +
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)